ddtrace/span.py

```diff
@@ -146,7 +146,7 @@ def __init__(
         # sampling
         self.sampled = True  # type: bool
 
-        self._context = context  # type: Optional[Context]
+        self._context = context._with_span(self) if context else None  # type: Optional[Context]
```

Comment:
Too bad we still need to copy contexts 🙁 . Would it be worth to add a comment here explaining why the `if ... else`? Or perhaps explain this in the PR description?

Reply:
Yeah, I meant to convey it with the first line of the PR:

> Each span needs its own context copy as the span.context property
returns the trace context specialized with the span's id.

Looking again at this change though it looks like we could maybe defer the copy till later but I can't imagine that the copy is expensive now that the lock is gone 🤔